module_auto_update/models/module.py

```diff
@@ -79,3 +70,8 @@ def update_list(self):
         )
         upgradeable_modules.write({'state': "to upgrade", })
         return res
+
+    @api.multi
+    def write(self, vals):
+        self._store_checksum_installed(vals)
+        return super(Module, self).write(vals)
```

Comment:
First write the record and then call the other method, or it could have undesired results.